src/gui/TimeLineWidget.cpp

```diff
@@ -304,60 +307,103 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
 
 
 
+void TimeLineWidget::contextMenuEvent(QContextMenuEvent*)
+{
+	QMenu contextMenu(tr("Timeline"), this);
+	QAction selectLoopPoints(tr("Select between loop points"), this);
+	connect(&selectLoopPoints, &QAction::triggered, this, [this](){
+		emit regionSelectedFromPixels(
+			m_xOffset + m_loopPos[0] * m_ppb / TimePos::ticksPerBar(),
+			m_xOffset + m_loopPos[1] * m_ppb / TimePos::ticksPerBar()
+		);
+		emit selectionFinished();
+	});
+	contextMenu.addAction(&selectLoopPoints);
+	contextMenu.exec( QCursor::pos() );
+}
+
+
+
+
 void TimeLineWidget::mousePressEvent( QMouseEvent* event )
 {
-	if( event->x() < m_xOffset )
-	{
-		return;
-	}
-	if( event->button() == Qt::LeftButton  && !(event->modifiers() & Qt::ShiftModifier) )
-	{
-		m_action = MovePositionMarker;
-		if( event->x() - m_xOffset < s_posMarkerPixmap->width() )
-		{
-			m_moveXOff = event->x() - m_xOffset;
-		}
+	// TODO: Read these from a config
+	auto leftCtrlAction = DragLoop;
+	auto rightCtrlAction = MoveLoopClosest;
+	auto leftShiftAction = MoveLoopBegin;
+	auto rightShiftAction = MoveLoopEnd;
+	auto button = event->button();
+	auto mods = event->modifiers();
+	
+	// Unmodified LMB is reserved for playhead, unmodified RMB for context menu
+	// Shift or Ctrl modified press behavior is bound by the user
+	// Shift + Ctrl modifier is reserved for fine adjustment of Shift actions
+	// TODO: Let MMB be bound
+	
+	if (event->x() < m_xOffset) { return; }
+	
+	// Choose an action based on input event and user's bindings
+	else if (button == Qt::LeftButton) {
+		if (mods & Qt::ShiftModifier){ m_action = leftShiftAction; }
+		else if (mods == Qt::ControlModifier){ m_action = leftCtrlAction; }
+		// Handles moving playhead
 		else
 		{
-			m_moveXOff = s_posMarkerPixmap->width() / 2;
+			m_action = MovePositionMarker;
+			if (event->x() - m_xOffset < s_posMarkerPixmap->width())
+			{
+				m_moveXOff = event->x() - m_xOffset;
+			}
+			else { m_moveXOff = s_posMarkerPixmap->width() / 2; }
 		}
 	}
-	else if( event->button() == Qt::LeftButton  && (event->modifiers() & Qt::ShiftModifier) )
+	else if (button == Qt::RightButton) {
+		if (mods & Qt::ShiftModifier){ m_action = rightShiftAction; }
+		else if (mods == Qt::ControlModifier){ m_action = rightCtrlAction; }
+		else { contextMenuEvent(nullptr); }
+	}
+	
+	// Calculate the clicked position as a TimePos, several actions need this
+	m_moveXOff = s_posMarkerPixmap->width() / 2;
```

Comment:
It does, but as far as I can tell moving the playhead still moves perfectly fine. Frankly I have no idea what `m_moveXOff` is supposed to do or in what edge case it ends up not being half the pixmap width. I've tried clicking within the playhead triangle, and clicking so that it's by the start of the song, and everything works.

Reply:
I see, it's probably not noticeable because the width is very small. We should probably decide between both ways though. If it's being overridden without apparent issue, maybe we should just delete the lines that calculate the offset when it's smaller than the width of the marker.